### PR TITLE
qt5-qtmultimedia: fix build in Xcode 15

### PR DESCRIPTION
**Pull request by markmentovai**
In Xcode 15 clang in C++17 mode, std::unary_function, deprecated since C++11, has been removed.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.0 23A344 arm64
Xcode 15.0 15A240d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

@MarcusCalhoun-Lopez

### CODE REVIEW

**Comment by chrstphrchvz** [comment]
Thanks for finding this upstream patch. However on https://bugreports.qt.io/browse/QTBUG-113782 it is not yet clear whether upstream is going to apply this to 5.15. I would like to verify that this still builds on macOS 10.13 with Xcode 9.4.1, which is pre-C++17.

File: aqua/qt5/Portfile [hunk 1]
```diff
@@ -1867,7 +1867,9 @@ foreach {module module_info} [array get modules] {
                 #    for linking.
                 # We must choose one or the other.
                 # see https://trac.macports.org/ticket/54592
-                patchfiles-append patch-qtmultimedia_no_macports_openal.diff
+                patchfiles-append \
+                    patch-qtmultimedia_no_macports_openal.diff \
+                    patch-qtmultimedia_no_std_unary_function.diff
```
Comment (chrstphrchvz) [inline aqua/qt5/Portfile:1872]: This `patchfiles-append` already has a comment not relevant to the patch being added, so I suggest not reusing it:
```suggestion
                patchfiles-append patch-qtmultimedia_no_macports_openal.diff

                # std::unary_function removed in Xcode clang 15
                patchfiles-append patch-qtmultimedia_no_std_unary_function.diff
```

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]

**Comment by chrstphrchvz** [comment]
A similar patch is likely needed for qt6-qtmultimedia, but I think it is okay to take care of that in another PR.

**Comment by chrstphrchvz** [comment]
Someone has since opened a ticket for this issue: https://trac.macports.org/ticket/68276

**Comment by markmentovai** [comment]
Thanks, I updated this to incorporate your feedback.

There’s nothing C++17-speciifc about this patch, so I see no reason it wouldn’t be compatible with any Xcode/clang/libc++ combination that this port already built under.

**Review by chrstphrchvz — APPROVED** [review]
Builds fine with Xcode clang 9.4.1 on macOS 10.13. (Although I would agree that getting Xcode 15 to work is the priority.)

**Pull request merged**